databuilder/databuilder/publisher/neo4j_csv_publisher_apoc.py

```diff
@@ -0,0 +1,462 @@
+        for node_file in self._node_files:
+            self._create_indices(node_file=node_file)
+
+        LOGGER.info('Publishing Node files: %s', self._node_files)
+        while True:
+            try:
+                node_file = next(self._node_files_iter)
+                LOGGER.info('_publish_node(%s)', node_file)
+                quick = time.time()
+                self._publish_node(node_file)
+                LOGGER.info('Successfully published nodes. Elapsed: %i seconds', time.time() - quick)
+            except StopIteration:
+                break
+
+        LOGGER.info('Publishing Relationship files: %s', self._relation_files)
+        while True:
+            try:
+                relation_file = next(self._relation_files_iter)
+                LOGGER.info('_publish_relation(%s)', relation_file)
+                quick = time.time()
+                self._publish_relation(relation_file)
+                LOGGER.info('Successfully published relationships. Elapsed: %i seconds', time.time() - quick)
+            except StopIteration:
+                break
+
+        # TODO: Add statsd support
+        LOGGER.info('Successfully published all. Elapsed: %i seconds', time.time() - start)
+
+    def get_scope(self) -> str:
+        return 'publisher.neo4j'
+
+    def _create_indices(self, node_file: str) -> None:
+        """
+        Go over the node file and try creating unique index
+        :param node_file:
+        :return:
+        """
+        LOGGER.info('Creating indices. (Existing indices will be ignored)')
+
+        with open(node_file, 'r', encoding='utf8') as node_csv:
+            for node_record in pandas.read_csv(node_csv, na_filter=False).to_dict(orient='records'):
+                label = node_record[NODE_LABEL_KEY]
+                if label not in self.labels:
+                    self._try_create_index(label)
+                    self.labels.add(label)
+
+        LOGGER.info('Indices have been created.')
+
+    def _chunker(self, seq, size):
+        return (seq[pos:pos + size] for pos in range(0, len(seq), size))
+
+    def _unquote(self, k) -> str:
+        if k.endswith(UNQUOTED_SUFFIX):
+            k = k[:-len(UNQUOTED_SUFFIX)]
+        return k
+
+    def _publish_node(self, node_file: str) -> None:
+        """
+        Iterate over the csv records of a file, each csv record transform to apoc.node.merge statement and will be executed.
+        All nodes should have a unique key, and this method will try to create unique index on the LABEL when it sees
+        first time within a job scope.
+        Example of Cypher query executed by this method:
+        CALL apoc.periodic.iterate(
+            'UNWIND $rows AS row RETURN row', 
+            'CALL apoc.merge.node([row.label], {key:row.key}, row, row) YIELD node RETURN COUNT(*);', 
+            {batchSize: $batch_size, iterateList: True, parallel: False, params: { rows: $batch, tag: $publish_tag }}
```

Comment:
```suggestion
            {batchSize: $batch_size, parallel: False, params: { rows: $batch, tag: $publish_tag }}
```

I would like to remove `iterateList` as it's deprecated and use the default `batchMode` which is `BATCH`, that has the same behaviour.